ome_zarr/writer.py

```diff
@@ -122,6 +122,23 @@ def _validate_plate_rows_columns(
     return validated_list
 
 
+def _validate_datasets(datasets: List[Union[str, dict]]) -> List[Dict]:
```

Comment:
With the proposed changes in 0.4 making `coordinateTransformations` mandatory, this API should likely be made `fmt` dependent with version-specific implementations delegated to the `Format` instances?